### PR TITLE
Output units when creating CODAP attributes

### PR DESCRIPTION
**Pull request by kswenson**
Output units when creating CODAP attributes [#152074847]

### CODE REVIEW

**Comment by scytacki** [comment]
@kswenson I merged master into this to fix some conflicts.  There was one part I was not sure about about. I'll add a line comment.

File: src/models/codap.ts [hunk 1]
```diff
@@ -212,38 +220,39 @@ export class Codap {
...
+        }
+
+        for(var i=0; i < sampleCount; i++) {
+            let item:IDataItem = {Run: this.state.runNumber };
+            for(j=0; j < 2; j++) {
+                let spec = dataSpecs[j],
+                    sample = spec.data[i];
+                if (item.Time == null)
+                    item.Time = sample[0];
+                if (sample[1] != null)
```
Comment (scytacki) [inline src/models/codap.ts]: I added this check to match what had been added to master from another PR. However, I wasn't sure if it was needed anymore.

**Comment by scytacki** [comment]
Well there is a typescript problem with this, due to the `Time` field of `IDataItem`. I'm not sure if it should be made optional, or perhaps it should be initialized to null or -1.

**Comment by kswenson** [comment]
Both merge changes look good. You may merge when ready.

**Comment by scytacki** [comment]
@kswenson there is the unresolved issues of the Time field on IDataItem.  This is why the build is failing. Do you have an opinion on which way we should address that?

**Comment by kswenson** [comment]
I think making `Time` optional is the simplest/cleanest. One could conceivably make the temporary `any` and then convert on assignment, but that seems like more trouble than it's worth. We're initializing the `IDataItem1 without a `Time` value with expectation of filling it in later, so in that sense it's optional.

**Pull request merged**